PowerShellEditorServices.build.ps1

```diff
@@ -55,28 +55,52 @@ function Invoke-WithCreateDefaultHook {
 
 function Install-Dotnet {
     param (
-        $Channel
+        [string[]]$Channel
     )
 
-    Write-Host "`n### Installing .NET CLI $Version...`n" -ForegroundColor Green
+    $env:DOTNET_INSTALL_DIR = "$PSScriptRoot/.dotnet"
+
+    Write-Host "Installing .NET channels $Channel" -ForegroundColor Green
 
     # The install script is platform-specific
     $installScriptExt = if ($script:IsUnix) { "sh" } else { "ps1" }
+    $installScript = "dotnet-install.$installScriptExt"
 
     # Download the official installation script and run it
-    $installScriptPath = "$([System.IO.Path]::GetTempPath())dotnet-install.$installScriptExt"
-    Invoke-WebRequest "https://dot.net/v1/dotnet-install.$installScriptExt" -OutFile $installScriptPath
-    $env:DOTNET_INSTALL_DIR = "$PSScriptRoot/.dotnet"
+    $installScriptPath = Join-Path ([System.IO.Path]::GetTempPath()) $installScript
+    Invoke-WebRequest "https://dot.net/v1/$installScript" -OutFile $installScriptPath
+
+    # Download and install the different .NET channels in parallel
+    foreach ($dotnetChannel in $Channel)
+    {
+        Write-Host "`n### Installing .NET CLI $Version...`n"
+
+        if ($script:IsUnix) {
+            chmod +x $installScriptPath
+        }
+
+        $params = if ($script:IsUnix)
+        {
+            @('-Channel', $dotnetChannel, '-InstallDir', $env:DOTNET_INSTALL_DIR, '-NoPath', '-Verbose')
+        }
+        else
+        {
+            @{
+                Channel = $dotnetChannel
+                InstallDir = $env:DOTNET_INSTALL_DIR
+                NoPath = $true
+                Verbose = $true
+            }
+        }
```

Comment:
why did you do this instead of what I had before with `Invoke-Expression`?

Reply:
Broke with my path which has spaces in it. Needing to fix it and wanting to avoid `iex` I went with this

Reply:
Fine I guess... annoying having a platform check for the same thing.

Reply:
cc @daxian-dbw

Reply:
```
> invoke-build build
Build build C:\Users\Robert Holt\Documents\Dev\Microsoft\PowerShellEditorServices\PowerShellEditorServices.build.ps1
Task /build/BinClean
Done /build/BinClean 00:00:00.0032695
Task /build/SetupDotNet

### Installing .NET CLI ...

ERROR: The term 'C:\Users\Robert' is not recognized as the name of a cmdlet, function, script file, or operable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
At C:\Users\Robert Holt\Documents\Dev\Microsoft\PowerShellEditorServices\PowerShellEditorServices.build.ps1:76 char:5
+     Invoke-Expression "$installScriptPath $paramArr"
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
At C:\Users\Robert Holt\Documents\Dev\Microsoft\PowerShellEditorServices\PowerShellEditorServices.build.ps1:82 char:1
+ task SetupDotNet -Before Clean, Build, TestHost, TestServerWinPS, Tes …
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
At C:\Users\Robert Holt\Documents\Dev\Microsoft\PowerShellEditorServices\PowerShellEditorServices.build.ps1:211 char:1
+ task Build BinClean,{
+ ~~~~~~~~~~~~~~~~~~~~~
Build FAILED. 3 tasks, 1 errors, 0 warnings 00:00:01.7445922
Invoke-Expression: C:\Users\Robert Holt\Documents\Dev\Microsoft\PowerShellEditorServices\PowerShellEditorServices.build.ps1:76
Line |
  76 |      Invoke-Expression "$installScriptPath $paramArr"
     |      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | The term 'C:\Users\Robert' is not recognized as the name of a cmdlet, function, script file, or
     | operable program. Check the spelling of the name, or if a path was included, verify that the path is
     | correct and try again.
```

Reply:
Oh I totally believe you... I just had exactly this code once and wanted to not have to do this, which is why I used iex in the first place. I would guess that if we wrapped InstallDir in quotes in might work... but seems hacky.

Reply:
Yeah, fixing it I didnt' want to get into a quote escaping tarpit. Ideally install-dotnet.ps1 would work cross-platform. Frankly that script could use some improvement anyway